### PR TITLE
[ACR] Support anon access in ACR blob client

### LINKED ISSUE #25083: [ACR] Support anonymous authentication for ContainerRegistryBlobClient
See 
* https://docs.microsoft.com/azure/container-registry/anonymous-pull-access
* https://github.com/Azure/bicep/issues/5031

### PR DESCRIPTION
**Pull request by Mohit-Chakraborty**

### CODE REVIEW

File: .vscode/cspell.json [hunk 1]
```diff
@@ -58,6 +54,7 @@
     "azsdk",
     "blazor",
     "contoso",
+    "deduplication",
```
Comment (annelo-msft) [inline .vscode/cspell.json:57]: Is this needed for ACR anon access?

File: eng/common/pipelines/templates/steps/sparse-checkout.yml [hunk 2]
```diff
@@ -38,8 +38,8 @@ steps:
             } else {
               Write-Host "Repository $($repository.Name) is being initialized."
 
-              Write-Host "git clone --no-checkout --filter=tree:0 git://github.com/$($repository.Name) ."
-              git clone --no-checkout --filter=tree:0 git://github.com/$($repository.Name) .
+              Write-Host "git clone --no-checkout --filter=tree:0 https://github.com/$($repository.Name) ."
```
Comment (annelo-msft) [inline eng/common/pipelines/templates/steps/sparse-checkout.yml:41]: Same question re: ACR anon access.

**Review by annelo-msft — APPROVED** [review]
Changes to the two ACR files LGTM

File: eng/common/pipelines/templates/steps/sparse-checkout.yml (hunk 2 above)
Comment (Mohit-Chakraborty) [reply eng/common/pipelines/templates/steps/sparse-checkout.yml:41]: Yeah, these changes are needed to keeping the tooling happy. CI builds now have additional checks and I brought the minimum changes in to pass the CI checks (cspell and sparse-checkout fixes).

File: .vscode/cspell.json (hunk 1 above)
Comment (Mohit-Chakraborty) [reply .vscode/cspell.json:57]: Yeah, these changes are needed to keeping the tooling happy. CI builds now have additional checks and I brought the minimum changes in to pass the CI checks (cspell and sparse-checkout fixes).

**Pull request merged**